source/MaterialXGenShader/ColorManagementSystem.h

```diff
@@ -73,6 +64,12 @@ class ColorManagementSystem
     /// Create a node to use to perform the given color space transformation.
     ShaderNodePtr createNode(const ColorSpaceTransform& transform, const string& name, ShaderGenerator& shadergen, const GenOptions& options);
 
+    /// Get a list of implementations used for shader generation
+    const std::unordered_map<string, ShaderNodeImplPtr>& getImplementationsUsed() const
+    {
+        return _cachedImpls;
+    }
```

Comment:
Could also use `getCachedImplementation` instead, see below.

Reply:
I'll make the same change here. But will require NlogN time for searching (as above).

Reply:
As above, complexity was NlogN already so this should not be worse, if I didn't miss something.

Reply:
Yes. I was thinking that there would be less searching and it's not so.